src/Microsoft.ML.Data/DataView/CacheDataView.cs

```diff
@@ -1419,8 +1423,8 @@ public ImplOne(CacheDataView parent, DataViewRowCursor input, int srcCol, Ordere
                     : base(parent, input, srcCol, waiter)
                 {
                     _getter = input.GetGetter<T>(input.Schema[srcCol]);
-                    if (parent._rowCount >= 0)
-                        _values = new T[(int)parent._rowCount];
+                    if (parent.GetRowCount().HasValue)
```

Comment:
```suggestion
                    if (parent.GetRowCount() is { } rowCount)
```

Reply:
Thanks Sam, interesting gramma

---
In reply to: [378953044](https://github.com/dotnet/machinelearning/pull/4829#discussion_r378953044) [](ancestors = 378953044)

Reply:
I am not familiar with this syntax. Can one of you please explain?

---
In reply to: [379041295](https://github.com/dotnet/machinelearning/pull/4829#discussion_r379041295) [](ancestors = 379041295,378953044)

Reply:
It's equivalent to `parent.GetRowCount() is int rowCount`, but infers the type of `int` from the left hand side. It is _not_ the same as `is var rowCount`, because `is var` does not perform a null check.